chromadb/test/property/strategies.py

```diff
@@ -1,32 +1,32 @@
 import hypothesis
 import hypothesis.strategies as st
-from typing import Dict, Optional, Sequence, TypedDict, cast
+from typing import Optional, Sequence, TypedDict, cast
 import hypothesis.extra.numpy as npst
-import numpy.typing as npt
 import numpy as np
 import chromadb.api.types as types
 import re
 
 # See Hypothesis documentation for creating strategies at
 # https://hypothesis.readthedocs.io/en/latest/data.html
 
-metadata = st.from_type(Optional[types.Metadata])
+collection_metadata = st.from_type(Optional[types.Metadata])
 
 # TODO: build a strategy that constructs english sentences instead of gibberish strings
-# Unsure what would happen feeding random unicode to an embedding model, could get bad results
 
-document = st.from_type(Optional[str])
-
-_coll_name_re = re.compile(r"^[a-zA-Z][a-zA-Z0-9-]{1,61}[a-zA-Z0-9]$")
+# TODO: collection names should be arbitrary strings
+# _collection_name_re = re.compile(r"^[a-zA-Z][a-zA-Z0-9-]{1,61}[a-zA-Z0-9]$")
+_collection_name_re = re.compile(r"^[a-z0-9][a-z0-9._-]{1,60}[a-z0-9]$")
```

Comment:
This is failing for me right now FYI due to an ID generated as '0000000'

Reply:
I was able to repro, and believe this is a bug. 

The string `0000000` is matched by the regex we use for IP address matching, which is currently `^[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}$`.

Why do we exclude this as a collection name? And should it be `^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$` (excaping the `.`) instead?

@levand

Reply:
The idea was that our naming semantics were "same as S3" and so that regex is trying to forbid collection names that are also valid IP addresses.

The regex is just wrong, the periods aren't escaped.

Reply:
Short-term fix is to escape the `.`, long term is to relax collection names validation so any string will work.